exoscale/api/compute.py

```diff
@@ -2824,129 +2991,147 @@ def create_instance_pool(
+        kwargs = {}
+
         if instance_anti_affinity_groups:
-            instance_anti_affinity_group_ids = [
-                i.id for i in instance_anti_affinity_groups
+            kwargs["anti-affinity-groups"] = [
+                {"id": i.id} for i in instance_anti_affinity_groups
             ]
 
-        instance_security_group_ids = None
+        if instance_elastic_ips:
+            kwargs["elastic-ips"] = [{"id": i.id} for i in instance_elastic_ips]
+
         if instance_security_groups:
-            instance_security_group_ids = [i.id for i in instance_security_groups]
+            kwargs["security-groups"] = [{"id": i.id} for i in instance_security_groups]
 
-        instance_private_network_ids = None
         if instance_security_groups:
-            instance_private_network_ids = [i.id for i in instance_private_networks]
+            kwargs["private-networks"] = [
+                {"id": i.id} for i in instance_private_networks
+            ]
 
-        instance_ssh_key_name = None
         if instance_ssh_key:
-            instance_ssh_key_name = instance_ssh_key.name
+            kwargs["ssh-key"] = instance_ssh_key.name
+
+        if instance_deploy_target:
+            kwargs["deploy-target"] = {"id": instance_deploy_target.id}
 
         instance_user_data_content = None
         if instance_user_data is not None:
             instance_user_data_content = b64encode(
                 bytes(instance_user_data, encoding="utf-8")
-            )
+            ).decode("ascii")
 
-        try:
-            res = self.cs.createInstancePool(
-                name=name,
-                description=description,
-                size=size,
-                zoneid=zone.id,
-                serviceofferingid=instance_type.id,
-                templateid=instance_template.id,
-                rootdisksize=instance_volume_size,
-                affinitygroupids=instance_anti_affinity_group_ids,
-                securitygroupids=instance_security_group_ids,
-                networkids=instance_private_network_ids,
-                keypair=instance_ssh_key_name,
-                userdata=instance_user_data_content,
-            )
-        except CloudStackApiException as e:
-            raise APIException(e.error["errortext"], e.error)
+        res = self._v2_request_async(
+            "POST",
+            "/instance-pool",
+            zone=zone.name,
+            json={
+                **{
+                    "description": description,
+                    "disk-size": instance_volume_size,
+                    "instance-prefix": instance_prefix,
+                    "instance-type": {"id": instance_type.id},
+                    "ipv6-enabled": instance_enable_ipv6,
+                    "name": name,
+                    "size": size,
+                    "template": {"id": instance_template.id},
+                    "user-data": instance_user_data_content,
+                },
+                **kwargs,
```

Comment:
```suggestion
            json={
                "description": description,
                "disk-size": instance_volume_size,
                "instance-prefix": instance_prefix,
                "instance-type": {"id": instance_type.id},
                "ipv6-enabled": instance_enable_ipv6,
                "name": name,
                "size": size,
                "template": {"id": instance_template.id},
                "user-data": instance_user_data_content,
                **kwargs,
```

There's a few other `**{…}` constructs that could be updated similarly. Sorry for noticing only now.